mlflow/gateway/__init__.py

```diff
@@ -0,0 +1,116 @@
+from multiprocessing import Process
+import pathlib
+import time
+from typing import Optional
+from uvicorn import Server, Config
+from watchdog.observers import Observer
+from watchdog.events import FileSystemEventHandler
+
+from mlflow.exceptions import MlflowException
+from mlflow.gateway.constants import _CONFIGURATION_FILE
+from mlflow.gateway.handlers import _load_gateway_config
+from mlflow.protos.databricks_pb2 import BAD_REQUEST
+
+
+class GatewayServer(Process):
+    """
+    Process management wrapper around the uvicorn server process to provide API-driven
+    control over the initialized process
+    """
+
+    def __init__(self, config: Config):
+        super().__init__()
+        self.config = config
+        self.server = Server(config)
+
+    def stop(self):
+        """
+        Stops the uvicorn server by terminating its process
+        """
+        self.terminate()
+
+    def run(self):
+        """
+        Starts the uvicorn server
+        """
+        self.server.run()
+
+
+class ConfigHandler(FileSystemEventHandler):
```

Comment:
Rework this. We don't need file monitors for MVP. Let's just save the current config to a tmp directory and attempt to restart with a fallback if an exception is raised during the server restart process from the passed-in pydantic config